plugin-server/src/main/utils.ts

```diff
@@ -63,11 +63,12 @@ export async function kafkaHealthcheck(
         let kafkaConsumerWorking = false
         let timer: Date | null = new Date()
         const waitForConsumerConnected = new Promise<void>((resolve) => {
-            consumer.on(consumer.events.FETCH_START, () => {
+            const removeListener = consumer.on(consumer.events.FETCH_START, () => {
```

Comment:
You can do this more idiomatically by just switching from `on` to `once` (https://nodejs.org/api/events.html#events_eventtarget_addeventlistener_type_listener_options)